pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/UuidPreProcessor.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import org.asciidoctor.ast.Document;
+import org.asciidoctor.extension.Preprocessor;
+import org.asciidoctor.extension.PreprocessorReader;
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import org.apache.sling.api.resource.ResourceResolver;
+import java.util.Optional;
+
+import org.apache.sling.api.resource.Resource;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UuidPreProcessor extends Preprocessor {   // (1)
+
+    private static Resource module;
+    private static final Logger log = LoggerFactory.getLogger(UuidPreProcessor.class);
+    private static String newModulePath;
+
+    public UuidPreProcessor(Resource module) {
+        this.module = module;
+    }
+
+    @Override
+    public void process(Document document, PreprocessorReader reader) {
+        
+        List<String> lines = reader.readLines(); 
+        List<String> newLines = new ArrayList<String>();
+        String[] split;
+        String uuid, newLink;
+        
+        for (String line: lines) {
+            if(line.startsWith("xref:")){
+                split = line.split(",pantheon-id=");
+                uuid = split[1].replace(split[1].substring(split[1].length()-1), "");                
+                resolveActualPath(module.getResourceResolver(), uuid);
+                newLink = split[0].replaceAll(":.*?\\[", ":"+newModulePath+"[");                
+                newLink = newLink + "]";
+                newLines.add(newLink);
```

Comment:
I would like to see this logic being conditional based on whether or not the UUID query returns a result. It looks like the current behavior is that if the UUID fails to resolve, the xref is totally wiped out and replaced with nothing. What the code _should_ do is leave the original author-supplied xref target intact.